extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/QuteProcessor.java

```diff
@@ -2191,6 +2197,14 @@ private boolean isApplicationArchive(ResolvedDependency dependency, Set<Applicat
         return false;
     }
 
+    static String readTemplateContent(Path path) {
+        try {
+            return Files.readString(path);
+        } catch (IOException e) {
+            throw new IllegalStateException("Unable to read the template content from path: " + path);
```

Comment:
Better propagate the IOException as the cause here? Also maybe `UncheckedIOException` may be more intuitive